src/table/utils/styling-utils.js

```diff
@@ -97,27 +103,37 @@ export function getBodyCellStyle(layout, theme) {
   // the same as whatever the font color is when the hover background
   // color is a light color.
   // 2. There is a hover font color but no hover background color,
-  // when hovering, only a hover font color is applied and the hover
-  // background color disappears.
+  // when hovering, only the font color is applied and no hover
+  // background color is shown.
   // 3. There is no hover font color and no hover background color,
-  // when hovering, the default hover effect (light gray background)
-  // 4. There are both hover font and background colors, when hovering,
-  // the hover font and background colors take effect.
-
-  const unsetHoverBackgroundColor = isUnset(content?.hoverColor);
-  const unsetHoverFontBackgroundColor = isUnset(content?.hoverFontColor) && unsetHoverBackgroundColor;
-
-  const setHoverBackgroundColor = unsetHoverBackgroundColor
-    ? ''
-    : getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, content?.hoverColor);
-
-  const hoverBackgroundColor = unsetHoverFontBackgroundColor
-    ? STYLING_DEFAULTS.HOVER_BACKGROUND
-    : setHoverBackgroundColor;
+  // when hovering, the default hover effect (a light gray hover background
+  // color and no hover font color).
+  // 4. There are both a hover font and a hover background color,
+  // when hovering, the hover font and the hover background color take effect.
+  // Note that hover colors from properties have a
+  // higher priority than those from theme.
+  const unsetHoverBackgroundColor = isUnset(contentHoverBackgroundColor) && !hoverBackgroundColorFromTheme;
+  const unsetHoverFontColor = isUnset(contentHoverFontColor) && !hoverFontColorFromTheme;
+  const unsetHoverFontBackgroundColor = unsetHoverBackgroundColor && unsetHoverFontColor;
+
+  // case 1 or 4
+  let hoverBackgroundColor = isUnset(contentHoverBackgroundColor)
+    ? hoverBackgroundColorFromTheme
+    : getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, contentHoverBackgroundColor);
+  hoverBackgroundColor = unsetHoverBackgroundColor
+    ? '' // case 2
+    : hoverBackgroundColor;
+  hoverBackgroundColor = unsetHoverFontBackgroundColor
+    ? STYLING_DEFAULTS.HOVER_BACKGROUND // case 3
+    : hoverBackgroundColor;
```

Comment:
Something like this should work if I understand it correctly:
1. there is a content hover color set -> use that
2. there is a hover color set on theme -> use that
3. If both 1 and 2 are false, no background is set. so now if a font color is set -> set to '' (no hover bg color)
4. else set to default color
```suggestion
  const unsetHoverFontColor = isUnset(contentHoverFontColor) && !hoverFontColorFromTheme;

  const  hoverBackgroundColor = !isUnset(contentHoverBackgroundColor)
    ? getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, contentHoverBackgroundColor) // case 1 and 4
    : hoverBackgroundColorFromTheme 
    ? hoverBackgroundColorFromTheme // case 1 and 4
    : !unsetHoverFontColor
    ? '' // case 2
    : STYLING_DEFAULTS.HOVER_BACKGROUND; // case 3

```

Reply:
Also I would change isUnset to isSet to remove the negation. Now we would be checking if stuff exists

Reply:
So without negation it should be like this
```suggestion
  const isHoverFontColorSet = isSet(contentHoverFontColor) || !!hoverFontColorFromTheme;
  const  hoverBackgroundColor = isSet(contentHoverBackgroundColor)
    ? getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, contentHoverBackgroundColor) // case 1 and 4
    : hoverBackgroundColorFromTheme 
    ? hoverBackgroundColorFromTheme // case 1 and 4
    : isHoverFontColorSet
    ? '' // case 2
: STYLING_DEFAULTS.HOVER_BACKGROUND; // case 3
```

Reply:
Learning from best practice, we should extract this nested ternary operation into an independent statement.

More to read:
https://medium.com/@benlmsc/stop-using-nested-ternary-operators-heres-why-53e7e078e65a

I gonna look into your 4 points and suggestion to change isUnset to isSet.

Reply:
I wouldn't call it best practice, there are pros and cons with both: https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340.
I'd say if statements are a bit easier to read, but I am fine with both as long as the logic is as simple as possible

Reply:
My code can also write like 
```
const hoverBackgroundColor = unsetHoverFontBackgroundColor
    ? STYLING_DEFAULTS.HOVER_BACKGROUND
    : unsetHoverBackgroundColor
    ? ''
    : isUnset(hoverBackgroundColorFromLayout)
    ? hoverBackgroundColorFromTheme
    : getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, hoverBackgroundColorFromLayout);
```
I think it is the same as yours except that isUnset is replaced with isSet.

Reply:
But for good readability, I dislike **nested** ternary operation

Reply:
I'm not saying you need to use nested terneries. I think it should be set only once. That you can do with if else and it would be easy to read as well